fairroot/base/steer/FairRun.cxx

```diff
@@ -82,6 +82,13 @@ FairRun::~FairRun()
 {
     LOG(debug) << "Enter Destructor of FairRun";
 
+    if (fSource) {
+        fSource->Close();
+    }
+    if (fSink) {
+        fSink->Close();
+    }
```

Comment:
In this case, we should just let program crash if `fSource` or `fSink` somehow is nullptr. 

```suggestion
        fSource->Close();
        fSink->Close();
```

Reply:
Well, FairRunSim does not have a Source. So it doesn't need to close it.

Reply:
I see. Yeah, that makes sense.